services/provider/server/server.go

```diff
@@ -810,11 +808,70 @@ func (s *OCSProviderServer) GetStorageClaimConfig(ctx context.Context, req *pb.S
 			}
 
 			// SID for RamenDR
-			storageID := calculateCephRbdStorageId(
+			storageID := calculateCephRbdStorageID(
 				cephCluster.Status.CephStatus.FSID,
 				strconv.Itoa(blockPool.Status.PoolID),
 				rns.Name)
 
+			if blockPool.GetAnnotations()[util.BlockPoolMirroringTargetIDAnnotation] != "" {
+				peerCephfsid, err := s.getPeerCephFSID(blockPool.Spec.Mirroring.Peers.SecretNames[0])
```

Comment:
1. You should not assume SecretNames is not empty
2. How do we know that we need to take the first item and not the second or the third?
3. Secret is located in the namespace of the block pool so you need to send the block pool namespace as well and don't assume it is the same namespace where the server lives
4. There is no reason the have `gerPeerCephFSID` as a method on the server if the only thing it needs from the server is the client. You should just send the client to the function as well

Reply:
Updated since we already know the secret name. If secret is not present we would log error and not create the VRCs